src/odemis/gui/comp/viewport.py

```diff
@@ -1065,7 +1066,334 @@ def _on_peak_method(self, state):
         pass
 
 
-class PointSpectrumViewport(PlotViewport):
+class NavigablePlotViewport(PlotViewport):
+    """ Abstract Class for displaying plotted data
+
+    Unlike PlotViewport, the NavigablePlotViewport can be manipulated by the user.
+    The plot can be zoomed and panned
+        - Mousewheel over an axis zooms that axis.
+        - Left mouse drag over an axis pans that axis.
+        - Mousewheel over the canvas zooms the horizontal axis
+        - Middle mouse button drag on the canvas pans the plot
+
+    This is controlled by VA's
+    .hrange: Tuple VA with the horizontal scale range
+    .vrange: Tuple VA with the vertical scale range
+    .hrange_lock, .vrange_lock: Bool VA that locks H/V axes from manipulation
+
+    Note: clamping of the ranges is handle by the mouse events, because
+    the clamping behavior must function differently for pans, drags, and scrolling
+
+    """
+    __metaclass__ = no_conflict.classmaker(right_metas=(ABCMeta,))
+
+    # Default class
+    canvas_class = miccanvas.NavigableBarPlotCanvas
+    bottom_legend_class = AxisLegend
```

Comment:
No need to override these two legend classes, as they are the same as in PlotViewport.